packages/vitest/src/node/viteLogger.ts

```diff
@@ -0,0 +1,133 @@
+import type { LogErrorOptions, LogLevel, LogType, Logger, LoggerOptions } from 'vite'
+import type { RollupError } from 'rollup'
+import colors from 'tinyrainbow'
+import type { Logger as VitestLogger } from './logger'
+
+const LogLevels: Record<LogLevel, number> = {
+  silent: 0,
+  error: 1,
+  warn: 2,
+  info: 3,
+}
+
+function clearScreen(logger: VitestLogger) {
+  const repeatCount = process.stdout.rows - 2
+  const blank = repeatCount > 0 ? '\n'.repeat(repeatCount) : ''
+  logger.clearScreen(blank)
+}
+
+let lastType: LogType | undefined
+let lastMsg: string | undefined
+let sameCount = 0
+
+// Only initialize the timeFormatter when the timestamp option is used, and
+// reuse it across all loggers
+let timeFormatter: Intl.DateTimeFormat
+function getTimeFormatter() {
+  timeFormatter ??= new Intl.DateTimeFormat(undefined, {
+    hour: 'numeric',
+    minute: 'numeric',
+    second: 'numeric',
+  })
+  return timeFormatter
+}
+
+export function createViteLogger(
+  console: VitestLogger,
```

Comment:
Ah, this is unfortunate, but copying looks fine :+1: 
Probably we should add the comment to the original source somewhere https://github.com/vitejs/vite/blob/main/packages/vite/src/node/logger.ts